From 8025980de920a26c83d79dabc3a6bde91422ac0e Mon Sep 17 00:00:00 2001
From: Xiexiaobo <X.Xie@freescale.com>
Date: Wed, 10 Jun 2009 10:28:31 +0800
Subject: [PATCH] ENGR00112477 SDHC: Remove GPIO pin inactive/active on suspend/resume

The MX25-3DS used a GPIO pin as detect_irq, when system suspend,
mxc_free_gpio() will disable this GPIO irq, but the mxc_request_gpio()
don't enable this GPIO irq, so the insertting and removing SD/MMC
can not be detected after suspend and resume operations. This fixed it.

Signed-off-by: Xie Xiaobo <X.Xie@freescale.com>
(cherry picked from commit 12fa5453b651c906a572792e39f6b667314ff723)
---
 drivers/mmc/host/mx_sdhci.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/drivers/mmc/host/mx_sdhci.c b/drivers/mmc/host/mx_sdhci.c
index fda0984..d8d53e6 100644
--- a/drivers/mmc/host/mx_sdhci.c
+++ b/drivers/mmc/host/mx_sdhci.c
@@ -1580,8 +1580,6 @@ static int sdhci_suspend(struct platform_device *pdev, pm_message_t state)
 		free_irq(chip->hosts[i]->irq, chip->hosts[i]);
 	}
 
-	gpio_sdhc_inactive(pdev->id);
-
 	return 0;
 }
 
@@ -1596,8 +1594,6 @@ static int sdhci_resume(struct platform_device *pdev)
 
 	DBG("Resuming...\n");
 
-	gpio_sdhc_active(pdev->id);
-
 	for (i = 0; i < chip->num_slots; i++) {
 		if (!chip->hosts[i])
 			continue;
-- 
1.5.4.4

